pkg/diagnose/diagnose.go

```diff
@@ -274,3 +363,299 @@ func (c *Collector) ExportDiagnostics(destDir string) {
+		return
+	}
+	if len(list.Items) == 0 {
+		collector.log.Printf("No NamespaceStores found.\n")
+		return
+	}
+
+	for _, namespacestore := range list.Items {
+		if namespacestore.Spec.Type != nbv1.NSStoreTypeNSFS {
+			analyzeNamespaceStore(cmd, &namespacestore, collector)
+		}
+	}
+}
+
+func analyzeBackingstore(cmd *cobra.Command, backingstore *nbv1.BackingStore, collector *Collector) {
+	analyzeResourceJob := loadAnalyzeResourceJob()
+	setImageInJob(analyzeResourceJob)
+	getBackingStoreDetailsForJob(backingstore, cmd, analyzeResourceJob)
+	setJobAnalyzeResource(cmd, analyzeResourceJob, collector)
+}
+
+func analyzeNamespaceStore(cmd *cobra.Command, namespaceStore *nbv1.NamespaceStore, collector *Collector) {
+	analyzeResourceJob := util.KubeObject(bundle.File_deploy_job_analyze_resource_yml).(*batchv1.Job)
+	analyzeResourceJob.Namespace = options.Namespace
+	setImageInJob(analyzeResourceJob)
+	getNamespacetoreDetailsForJob(namespaceStore, cmd, analyzeResourceJob)
+	setJobAnalyzeResource(cmd, analyzeResourceJob, collector)
+}
+
+func loadAnalyzeResourceJob() *batchv1.Job {
+	analyzeResourceJob := util.KubeObject(bundle.File_deploy_job_analyze_resource_yml).(*batchv1.Job)
+	analyzeResourceJob.Namespace = options.Namespace
+	return analyzeResourceJob
+}
+
+func collectorInstance(cmd *cobra.Command) *Collector {
+	kubeconfig, _ := cmd.Flags().GetString("kubeconfig")
+	collector := &Collector{
+		folderName:  fmt.Sprintf("%s_%d", "noobaa_analyze_resource", time.Now().Unix()),
+		log:         util.Logger(),
+		kubeconfig:  kubeconfig,
+		kubeCommand: util.GetAvailabeKubeCli(),
+	}
+	return collector
+}
+
+func makeDirForLogs(collector *Collector) {
+	err := os.Mkdir(collector.folderName, os.ModePerm)
+	if err != nil {
+		collector.log.Fatalf(`❌ Could not create directory %s, reason: %s`, collector.folderName, err)
+	}
+}
+
+func setImageInJob(analyzeResourceJob *batchv1.Job) {
+	noobaa := util.KubeObject(bundle.File_deploy_crds_noobaa_io_v1alpha1_noobaa_cr_yaml).(*nbv1.NooBaa)
+	noobaa.Namespace = options.Namespace
+	if !util.KubeCheck(noobaa) {
+		util.Logger().Fatalf(`❌ Could not get noobaa %q in Namespace %q`,
+			noobaa.Name, noobaa.Namespace)
+	}
+	analyzeResourceJob.Spec.Template.Spec.Containers[0].Image = noobaa.Status.ActualImage
+}
+
+func getBackingStoreDetailsForJob(backingstore *nbv1.BackingStore, cmd *cobra.Command, analyzeResourceJob *batchv1.Job) {
+	log := util.Logger()
+
+	// set type of backingstore
+	storeType := util.GetEnvVariable(&analyzeResourceJob.Spec.Template.Spec.Containers[0].Env, "RESOURCE_TYPE")
+	if storeType == nil {
+		log.Fatalf("❌ Could not get resource type")
+	}
+	storeType.Value = string(backingstore.Spec.Type)
+
+	// get bucket to analyze (uses the target bucket by default)
+	bucket := util.GetEnvVariable(&analyzeResourceJob.Spec.Template.Spec.Containers[0].Env, "BUCKET")
+	if bucket == nil {
+		log.Fatalf("❌ Could not get bucket")
+	}
+	bucketArg, _ := cmd.Flags().GetString("bucket")
+	if len(bucketArg) != 0 {
```

Comment:
is `bucket` an array? if its a string can we just use `if bucketArg == ""` check.

Reply:
I talked with @dannyzaken I will remove the option to type bucket name, it will only work on target bucket